pkg/query/querier.go

```diff
@@ -174,17 +182,18 @@ func (q *querier) Select(params *storage.SelectParams, ms ...*labels.Matcher) (s
 
 	resp := &seriesServer{ctx: ctx}
 	if err := q.proxy.Series(&storepb.SeriesRequest{
-		MinTime:             q.mint,
-		MaxTime:             q.maxt,
-		Matchers:            sms,
-		MaxResolutionWindow: q.maxSourceResolution,
-		Aggregates:          queryAggrs,
+		MinTime:                 q.mint,
+		MaxTime:                 q.maxt,
+		Matchers:                sms,
+		MaxResolutionWindow:     q.maxSourceResolution,
+		Aggregates:              queryAggrs,
+		PartialResponseDisabled: !q.partialResponse,
```

Comment:
We cannot if we really want to not break compatibility on gRPC storeAPI level.

Because right now partialResponse enabled is true, adding new `PartialResponseEnabled` will default to false when not specified (for example if we add this field and rollout leaf querier only, it will behave differently until we rollout top querier that will feed this field.)